### PR TITLE
Add basic documentation and setup RTD builds

### PR DESCRIPTION
**Pull request by itziakos**
This PR adds basic documentation for the current PyWin32 compatibility layer and sets up read-the-docs to automatically build and publish the documentation

### CODE REVIEW

**Comment by itziakos** [comment]
@cournape  one more :)

File: README.rst [hunk 1]
```diff
@@ -4,28 +4,40 @@
...
-    win32api.LoadLibraryEx(sys.executable, 0, win32api.LOAD_LIBRARY_AS_DATAFILE)
+  win32api.LoadLibraryEx(sys.executable, 0, win32api.LOAD_LIBRARY_AS_DATAFILE)
 
-Note: this implements only a very small subset of pywin32, for internal needs
-at Enthought. We do welcome additional features and prs, though.
+.. note::
+
+   Currently pywin32ctypes implements only a very small subset
+   of pywin32, for internal needs at Enthought. We do welcome
+   additional features and prs, though.
```
Comment (cournape) [inline README.rst]: that should be `PRs` I guess ?

File: README.rst [hunk 2]
```diff
@@ -4,28 +4,40 @@
...
 
 Development setup
 =================
 
 The following should be good enough::
 
-	pip install -r dev_requirements.txt
-	python setup.py develop
+  pip install -r dev_requirements.txt
+  python setup.py develop
```
Comment (cournape) [inline README.rst]: let's replace this w/ `pip install -e`, as this is the recommended way nowadays

File: README.rst [hunk 3]
```diff
@@ -4,28 +4,40 @@
...
+
+.. note::
 
-Note: because of the pywin32 dependency for tests, you most likely want to
-create a virtualenv with --system-site-packages if you use virtualenv.
+   - Because of the pywin32 dependency for tests, you most likely want to
+     create a virtualenv with --system-site-packages if you use virtualenv.
```
Comment (cournape) [inline README.rst]: This is actually incorrect thanks to `pywin32-pypi` package.

File: docs/source/conf.py [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+author = u'David Cournapeau, Ioannis Tziakos'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = '0.0'
+# The full version, including alpha/beta/rc tags.
+release = '0.0.2dev1'
```
Comment (cournape) [inline docs/source/conf.py]: Shouldn't this be populated automatically from the package ?

**Comment by cournape** [comment]
A few cosmetic comments, LGTM otherwise

**Pull request merged**